crt_portal/static/js/dashboard_view_all.js

```diff
@@ -1,42 +1,52 @@
 (function(root, dom) {
-  var toggleAll = dom.querySelector('a.td-toggle-all');
+  function toggleTarget(toggle) {
+    const arrow = toggle.querySelector('.icon');
+    const id = toggle.dataset['id'];
+    const summary = dom.getElementById(`tr-additional-${id}`);
+    arrow.classList.toggle('rotate');
+    summary.toggleAttribute('hidden');
+  }
+
+  function isToggled(target) {
+    return target.querySelector('.icon').classList.contains('rotate');
+  }
+
+  function markAsViewed(ids) {
+    if (!ids.length) return;
+    window.onbeforeunload = function() {
+      return 'Report views have not yet saved, are you sure you want to leave?';
+    };
+    window
+      .fetch(`/api/reports/`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-CSRFToken': Cookies.get('csrftoken')
+        },
+        mode: 'same-origin',
+        body: JSON.stringify({ report_pks: ids })
```

Comment:
Is this backwards compatible with the original endpoint?  We still use that for individual carrots, right?

Reply:
We do still use the original endpoint for single carrots.

The magic is in cts_forms/views.py: `def mark_reports_as_viewed(reports, user):
    for report in reports:
        mark_report_as_viewed(report, user)`

@okkays added this so that we can easily iterate over a list of primary keys and call 'mark_report_as_viewed' on each pk.